docs/docsite/rst/playbooks_tags.rst

```diff
@@ -86,27 +98,43 @@ You can apply tags to more than tasks, but they ONLY affect the tasks themselves
       tasks:
         ...
 
-You may also apply tags to roles::
+You may also apply tags to ``roles:``:
 
     roles:
       - { role: webserver, port: 5000, tags: [ 'web', 'foo' ] }
 
-And import statements::
+And to ``import_role:`` and ``import_tasks:`` statements::
+
+    - import_role:
+        name: myrole
+      tags: [web,foo]
 
     - import_tasks: foo.yml
       tags: [web,foo]
 
+
 All of these apply the specified tags to EACH task inside the play, imported
 file, or role, so that these tasks can be selectively run when the playbook
 is invoked with the corresponding tags.
 
-The above information does not apply to `include_tasks` or other dynamic includes,
-as the attributes applied to an include, only affect the include itself.
+Tags are inherited *down* the dependency chain. In order for tags to be
+applied to a role and all its dependencies, the tag should be applied to the
+role, not to all the tasks within a role.
+
+You can see which tags are applied to tasks and imported tasks by running
+``ansible-playbook`` with the ``--list-tasks`` option. You can display all
+tags applied to the tasks with the ``--list-tags`` option.
+
+.. note::
+    The above information does not apply to `include_tasks`, `include_roles`,
+    or other dynamic includes. Tags applied to either of these only tag the
+    include itself.
 
-Tags are inherited *down* the dependency chain. In order for tags to be applied to a role and all its dependencies,
-the tag should be applied to the role, not to all the tasks within a role.
+    To use tags with tasks and roles intended for dynamic inclusions,
+    all needed tasks should be explicitly tagged at the task level; or
+    ``block:`` may be used to tag more than one task at once. The include
+    itself should also be tagged.
```

Comment:
i think we probably need examples, i know what you mean and i still have hard time mapping it to a playbook

Reply:
What really bothers me about that advice is that it means that for any ansible-galaxy role one would want to loop over (or any role, of course), we're having to suggest tagging every task in the role with 'the same name as the role' to get 2.4 behavior, or indenting every task just to add the tag which again, is the same name as the role. But that's a problem for the include_ pages, I suppose.

Reply:
Added an example of a dynamic include (include_role) and a block statement in the role tasks file to tag all of the tasks in the role.